pkg/controller/backup/clusterdeployment.go

```diff
@@ -0,0 +1,169 @@
+	hasChanged, err := r.hasClusterDeploymentChanged(cd)
+	if err != nil {
+		// The only thing the watch handler can do is to log the error.
+		contextLogger.WithError(err).Error("Couldn't determine if cluster deployment had changed.")
+		return []reconcile.Request{}
+	}
+
+	if !hasChanged {
+		// Nothing has changed, do not queue.
+		return []reconcile.Request{}
+	}
+
+	// Queue up the NS for this ClusterDeployment
+	retval := []reconcile.Request{
+		{NamespacedName: types.NamespacedName{
+			Namespace: cd.Namespace,
+		}}}
+	return retval
+}
+
+func (r *ClusterDeploymentBackup) getChangedInNamespace(namespace string) ([]*hivev1.ClusterDeployment, error) {
+	contextLogger := addNamespaceLoggerFields(r.reconciler.logger, namespace)
+
+	allClusterDeployments := &hivev1.ClusterDeploymentList{}
+	err := r.reconciler.List(context.TODO(), allClusterDeployments, client.InNamespace(namespace))
+	if err != nil {
+		contextLogger.WithError(err).Error("Couldn't get list of ClusterDeployments")
+		return nil, err
+	}
+
+	changedClusterDeployments := []*hivev1.ClusterDeployment{}
+	for i, clusterDeployment := range allClusterDeployments.Items {
+		hasChanged, err := r.hasClusterDeploymentChanged(&clusterDeployment)
+		if err != nil {
+			return nil, err
+		}
+
+		if hasChanged {
+			// This one is different!
+			changedClusterDeployments = append(changedClusterDeployments, &allClusterDeployments.Items[i])
+		}
+	}
+
+	return changedClusterDeployments, nil
+}
+
+func (r *ClusterDeploymentBackup) hasClusterDeploymentChanged(clusterDeployment *hivev1.ClusterDeployment) (bool, error) {
+	contextLogger := addClusterDeploymentLoggerFields(r.reconciler.logger, clusterDeployment)
+
+	newChecksum, err := r.checksum(clusterDeployment)
+	if err != nil {
+		contextLogger.WithError(err).Info("error calculating checksum of cluster deployment")
+		return false, err
+	}
+
+	// Make sure newChecksum is different than the old checksum.
+	oldChecksum := clusterDeployment.Annotations[controllerutils.LastBackupAnnotation]
+	return newChecksum != oldChecksum, nil
+}
+
+func (r *ClusterDeploymentBackup) checksum(clusterDeployment *hivev1.ClusterDeployment) (string, error) {
+	// We need to take the checksum WITHOUT the previous checksum
+	// to avoid an infinite loop of always changing the checksum.
+	metaCopy := clusterDeployment.ObjectMeta.DeepCopy()
+	delete(metaCopy.Annotations, controllerutils.LastBackupAnnotation)
+
+	// We also need to take the checksum WITHOUT the previous ResourceVersion
+	// as when we update the checksum annotation, we see a new change.
+	metaCopy.ResourceVersion = ""
+
+	return r.checksumFunc(metaCopy, clusterDeployment.Spec)
+}
+
+func (r *ClusterDeploymentBackup) updateLastBackupChecksum(clusterDeployments []*hivev1.ClusterDeployment) error {
+	errors := []error{}
+
+	for _, clusterDeployment := range clusterDeployments {
+		clusterDeployment, err := controllerutils.
+			NewClusterDeploymentBuilderWithClusterDeployment(r.checksum, clusterDeployment).
```

Comment:
I am still not 100% sold on this approach here. The language makes it sounds like a new clusterdeployment is being built rather than the existing clusterdeployment being modified. And it is bit wordy compared to something like the following--which also has the benefit of being pretty clear about what it is doing.

```
controllerutils.SetBackupChecksum(clusterDeployment, r.checksum)
```